### PR TITLE
Improve syntax highlighting colors

### PR DESCRIPTION
**Pull request by lydell**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | no
| Deprecations?     | no
| Spec compliancy?  | no
| Tests added/pass? | yes
| Fixed tickets     | -
| License           | MIT
| Doc PR            | -

- Highlight the error location markers in bold red.
- Dim the line number gutter with grey.
- Don't highlight brackets. Few editor color schemes do.
- Improve JSX highlighting.
- Highlight capitalized variable names.
- Make invalid tokens stand out with a red background.
- Use a less noisy color theme, mostly based on shades of yellow.

Before:

screenshot from 2016-09-26 21-49-27

After:

screenshot from 2016-09-26 22-25-21

### CODE REVIEW

**Comment by hzoo** [comment]
> Make invalid tokens stand out with a red background. 👍

**Comment by hzoo** [comment]
Not really sure about changing to yellow or the bold keywords? Most of this is personal preference

**Comment by lydell** [comment]
> Most of this is personal preference

It sure is. What I can say is:

- I'm not much of a designer.
- I didn't like the old highlighting.
- I tried out _lots_ of different styles before settling on this one.
- It's hard to make a nice theme with the limited palette that chalk provides.
- I did copy my personal text editor theme a bit.
- I wanted the result to be easy to read, nice and not too noisy.
- I'm open for other suggestions regarding the color!

At least it's better than before, though?

**Comment by hzoo** [comment]
I think I'd be 👍  with the other changes. I think we can merge that part first and then make the other change a separate PR for further discussion?

> Use a less noisy color theme, mostly based on shades of yellow.

There's probably a lot of people who are used to the old one so if we are going to change it maybe should get more feedback before it's changed like that?

**Comment by lydell** [comment]
> I think we can merge that part first and then make the other change a separate PR for further discussion?

That's a good idea, but I don't know how to do that.

> There's probably a lot of people who are used to the old one so if we are going to change it maybe should get more feedback before it's changed like that?

Definitely! :)

**Comment by hzoo** [comment]
> That's a good idea, but I don't know how to do that.

So I would revert the change for `string:     chalk.red` for yellow and others if that's possible? maybe

```js
let defs = {
  string:     chalk.red,
  punctuator: chalk.bold,
  curly:      chalk.green,
  parens:     chalk.blue.bold,
  square:     chalk.yellow,
  keyword:    chalk.cyan,
  number:     chalk.magenta,
  regex:      chalk.magenta,
  comment:    chalk.grey,
  invalid:     chalk.white.bgRed.bold,
  
  // new stuff
  marker:      chalk.red.bold,
  gutter:      chalk.grey,
  // bracket:  intentionally omitted.
  capitalized: chalk.yellow,
  jsx_tag:     chalk.yellow.dim,
};
```

File: packages/babel-code-frame/src/index.js [hunk 1]
```diff
@@ -7,16 +7,18 @@ import chalk from "chalk";
...
+  jsx_tag:     chalk.yellow.dim,
+  punctuator:  chalk.yellow.dim,
+  // bracket:  intentionally omitted.
+  number:      chalk.yellow,
+  string:      chalk.yellow,
+  regex:       chalk.yellow,
+  comment:     chalk.grey,
+  invalid:     chalk.white.bgRed.bold,
+  gutter:      chalk.grey,
+  marker:      chalk.red.bold,
```
Comment (lydell) [inline packages/babel-code-frame/src/index.js]: Anyone want to play around with these colors? All you need to do is edit this block of code!

Quick-start:

1. `git clone https://github.com/babel/babel.git`
2. `make bootstrap`
3. Edit `packages/babel-code-frame/lib/index.js`.
4. Try it out with `node test.js`.
5. Post your changes and a screen shot here.

```
// test.js
const codeFrame = require('./packages/babel-code-frame/lib/')

const lines = `
class MyComponent extends React.Component {
  render() {
    const format = (title) => title.replace(/,$/, '') + 1337;
    return (
      <h1 className="test">
        {format(this.props.title)}
      </h1>
    );
  } // comment
} # invalid
`.trim()
console.log(codeFrame(lines, 6, 10, {highlightCode: true}))
```

Comment (hzoo) [inline packages/babel-code-frame/src/index.js]: I just used a babel project, wrote the code you pasted before and then ran it by editing node_modules

**Comment by lydell** [comment]
Wow, the colors look much nicer on your computer!

Note that the following rules you added are not applied in your screenshot (I guess you didn't change back to using different token types for every type of bracket):

```js
  curly:      chalk.green,
  parens:     chalk.blue.bold,
  square:     chalk.yellow,
```

**Comment by lydell** [comment]
What does my version look like on your computer?

**Comment by hzoo** [comment]
I'm using https://github.com/voronianski/oceanic-next-color-scheme in sublime and in iTerm so that could be a reference

**Comment by hzoo** [comment]
screen shot 2016-09-26 at 4 57 24 pm

The green/blue isn't that visible, seems good

**Comment by lydell** [comment]
- The thing I dislike the most about the old highlighting is the multi-colored brackets.
- The thing I dislike the second most about the old highlighting is the bold punctuators. They look pretty good in you screen shots, but not on Ubuntu. (See the first screen shot.)
- I like it when punctuator and jsx_tag have the same appearance, making `</tag>` styled as a single chunk.
- I don't think it's a good idea to have both strings and error markers be red. Red for strings also doesn't look very nice by default on Ubuntu.

Here's another alternative:

screenshot from 2016-09-26 23-19-55

**Comment by hzoo** [comment]
Yeah I think the changes to brackets to remove color were fine. Having red as string is odd but the background error color is red so it's not too bad

Ok but are React Components the other yellow? it seemed like before it was the dim one even when jsx was caps

**Comment by lydell** [comment]
> Yeah I think the changes to brackets to remove color were fine.

OK.

> Having red as string is odd but the background error color is red so it's not too bad

Sorry, I meant the `>` and `^` markers (not invalid tokens (such as `#`), which are _very_ uncommon).

> Ok but are React Components the other yellow? it seemed like before it was the dim one even when jsx was caps

The idea is that `<h1` and `<MyComponent` always should be dimmed yellow, while `MyComponent` is "regular" yellow. Have you got some other idea?

**Comment by hzoo** [comment]
Basically what I had in https://github.com/babel/babel/pull/4572#issuecomment-249692253 was keep everything the same and introduce less colors (use one yellow for caps/jsx) as a initial pr. And graying out the brackets does that as well

**Comment by lydell** [comment]
The "initial pr" does _not_ use one yellow for both caps/jsx. Look more closely at the "After" screen shot in the OP :)

Is this what you're looking for? :)

screenshot from 2016-09-27 07-12-35

**Comment by hzoo** [comment]
Yeah that's what I meant - it was 2 different colors but we could just use 1?

**Comment by lydell** [comment]
Sorry, I don’t really get it – is my last screen shot fine, or do I need to make more adjustments? :)

**Comment by hzoo** [comment]
Yeah 👍

**Comment by lydell** [comment]
@hzoo I've updated the PR now, so that it looks like in the last screen shot.

**Comment by hzoo** [comment]
screen shot 2016-09-27 at 10 35 47 am

Ok looks like ^ for me

**Comment by lydell** [comment]
That looks like expected to me.

Just tell me if there's anything more you'd like to be changed! :)

**Comment by hzoo** [comment]
Nope looks good! Dono if we want to add the picture in the readme or not

**Comment by lydell** [comment]
The _only_ thing _I_ would like to change now, is the red color for strings. Red makes it look like something is wrong with the string. And on Ubuntu, it looks kinda broken.

Now:

screenshot from 2016-09-27 16-29-34

Suggestion:

screenshot from 2016-09-27 16-30-04

Is this worth discussing? Should I open a PR?

**Comment by hzoo** [comment]
Sounds good, it's the same as oceanic in my editor

**Comment by lydell** [comment]
Ok, I submitted #4579 for the color change.

**Pull request merged**